### ISSUE TITLE
JS Engine Hermes: Typeerror: Cannot read property 'createNode' of null

### ISSUE THREAD

**Issue by tushargoyalofficial** [oblador/react-native-animatable#304]
We have a project over RN v0.61.5. We are using React Navigation v4. We are facing this error while using your library. Please help me out.

![Screenshot 2020-03-09 at 1 15 14 PM](https://user-images.githubusercontent.com/24288879/76192660-5d222280-6208-11ea-8299-169cd5c0e71a.png)

**Comment by nadunliyanage** [comment]
Did you find any solution for this? I am having this issue.

**Comment by PrasadGhan** [comment]
I am also facing the same issue.

**Comment by nikolai-git-house** [comment]
Me too.

**Comment by nikolai-git-house** [comment]
Any solution here?

**Comment by Nirony** [comment]
me 2, any news?

**Comment by saravanakumargn** [comment]
try to clear the cache.
```
npx react-native start --reset-cache
```

**Comment by bhavesh18** [comment]
Same issue!!

**Comment by chagai95** [comment]
getting the same error, when adding webrtc code that works on the browser... found this issue by googling the error, anyone understand it?

**Comment by tushargoyalofficial** [comment]
the solution was the correct native configuration in Java files. as mentioned in docs of the router flux

**Comment by tushargoyalofficial** [comment]
This issue comes when gesture handler is not configured correctly in the RN app for android.

**Comment by chagai95** [comment]
Can you link to these places or can I easily find them on google? I don't wanna take the wrong stuff thanks! Are you sure about this? are there any other possible reasons? I get it when I add my type script files to a non type script react native app and also when I add the generated JavaScript files from these typescript files...Thanks again!

**Comment by bona-ws** [comment]
Same issue here ... Any one know why it happens, and what is the solution ?

**Comment by grafosecondo** [comment]
+1

@tushargoyalofficial checked the the gesture handler configuration again and again

**Comment by hasgardee** [comment]
any fix?

**Comment by Divyanshu-Grover** [comment]
This is caused due to incorrect import for Easing animation. Please make sure it is added under 'react-native'. Auto importing Easing causes this error.

**Comment by bluescorp1311** [comment]
> This is caused due to incorrect import for Easing animation. Please make sure it is added under 'react-native'. Auto importing Easing causes this error.

Please tell me more about fixing this issue. How to import Easing animation? Thank you.

**Comment by kiranjd** [comment]
Deleting `node_modules` and running `yarn`(or `npm install`) worked for me

**Comment by Ahmed-Imam** [comment]
getting the same issue after migrating from v0.59.0 to 0.60.5, it occurs on Android only. 
deleting node_modules and reset cache didn't solve the issue. 

` "react-native": "0.60.5",
   "react-native-gesture-handler": "1.10.3",
   "react-navigation": "^3.0.9",
`

**Comment by mitrukahitesh** [comment]
The problem is with with react-native-reanimated. Follow the procedure as in
[](https://docs.swmansion.com/react-native-reanimated/docs/installation/)
It should work fine now. If it doesn't, try fixing react-native-gesture-handler as in
https://docs.swmansion.com/react-native-gesture-handler/docs/#installation

**Comment by jdhl27** [comment]
The error is very dynamic.
Check your main configuration files that everything is ok. In my case, in the `.env` I had an incorrect value that was needed for the state of the app.

**Comment by trinachaudhuri-ctri** [comment]
is this issue fixed, am facing the same, have tried all the above mentioned fixes but did not work

**Comment by ummartoor** [comment]
Same :(

**Comment by ummartoor** [comment]
Hi, Thanks for your answer. I've found the issue. I am using react-native-render-html and I updated that package too. Downgraded the package version to ^5.1.1 and now this is working fine. Seems like latest version of the react-native-render-html is not supporting with my current hermes configurations.
no idea if this one can help

**Comment by ummartoor** [comment]
https://stackoverflow.com/questions/58131602/unable-to-resolve-module-react-native-reanimated

Try this one for this error but after that you'll get other errors but at least you'll the point and hope so you'll find a way to remove it i am working on it i will let you know

**Comment by 77TecShaeer** [comment]
+++

**Comment by taranga99** [comment]
In my project it was cause by react native reanimated which i had to install while using the drawer in react navigation. All I had to do was to add reanimated's babel plugin in my babel.config file and did a cache reset to run the app. The instructions to add the plugin and reset cache is given in this link.
https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/installation/

**Comment by ryyvv** [comment]
remove this one in Index.js (react native) ->  import 'react-native-gesture-handler';

**Comment by SmartArray** [comment]
In my case the emulator was just not able to connect to the bundler.

Make sure `adb` is installed and the path variable is set up correctly. If needed, run `adb reverse tcp:8081 tcp:8081` to forward the port

**Comment by Xlizer1** [comment]
> the solution was the correct native configuration in Java files. as mentioned in docs of the router flux

what should i do, cuz im having the same error

**Comment by strawberryCheeseCake2** [comment]
idk how it works but here's what i've tried and fixed it (I was using stack navigator)
and I was working on iOS

1. delete node_modules
2. npm i -f
3. npm install @react-native-masked-view/masked-view
4. npx pod-install ios

**Comment by ryyvv** [comment]
Check this

[image: image.png]

**Comment by ryyvv** [comment]
if the emulator is not working properly

**Comment by betomoedano** [comment]
## Solution ✅

1. Make sure you added reanimated to your `babel.config.js` like this:
```
module.exports = {
  presets: ['module:metro-react-native-babel-preset'],
  plugins: ['react-native-reanimated/plugin'],
};
```
2. After updating your `babel.config.js` restart your server with the following command:
```
yarn start --resetCache
```
3. Finally, delete your App from your simulator. Then run on Android by pressing `a` in your terminal

![Screenshot 2023-02-18 at 12 24 09 PM](https://user-images.githubusercontent.com/43630417/219879603-89b8eb09-4f5e-4be2-a763-afbad8fc5741.png)

I hope this helps.

**Comment by ryyvv** [comment]
Awesome, thanks!

**Comment by iamrexan** [comment]
`getName` method should be equal to your java module class name, validated in RN 0.7+

`public String getName() {
        return "VpnSelectorModule";
    }`

**Comment by dansau84** [comment]
any help on this persistant error that nobody helped so far ? many users affected!!!   expo native react project

**Comment by cph2117** [comment]
The following solved it for me:
`npx expo install react-native-gesture-handler react-native-reanimated react-native-screens react-native-safe-area-context @react-native-community/masked-view`
then `import 'react-native-gesture-handler';` above all other imports in `App.js`.

It's required by React Native Navigation as detailed [here](https://reactnavigation.org/docs/5.x/getting-started/), but for whatever reason only breaks when it's built.

**Comment by Hardly002** [comment]
Check imports! For me, the problem was that I imported react-native elements from react.

**Comment by tribeless** [comment]
I fixed it by upgrading the two packages to:
"react-native-image-picker": "^7.0.1",
"react-native-reanimated": "^3.5.4",

**Comment by bugra-kara** [comment]
> I fixed it by upgrading the two packages to:
> "react-native-image-picker": "^7.0.1",
> "react-native-reanimated": "^3.5.4",

it worked for me, thanks!

**Comment by NewvilleMedia** [comment]
This worked for me
npm install @react-navigation/native-stack@latest 
then use Xcode to clean build.

I asked ChatGPT for this and got the solution

**Comment by PrabhathR-99** [comment]
- If you are using ios or Mac make sure that you have run pod install in the ios folder 
`cd ios
  pod install
  cd ..`

- Try to go with 'react-native-gesture-handler": "^2.8.0"', may be the new version is not working with your project, (npm install react-native-gesture-handler@2.8.0).

**Comment by brawler2011** [comment]
any help
```
This error is located at:
    in App
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
    in second(RootComponent), js engine: hermes
 ERROR  TypeError: Cannot read property 'lockToLandscape' of null

This error is located at:
    in App
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
    in second(RootComponent), js engine: hermes
```

**Comment by Ali136384** [comment]
> Did you find any solution for this? I am having this issue.

yes,the problem is from a library I`ve installed, so just check the libraries u using .

**Comment by maztapee** [comment]
> Deleting `node_modules` and running `yarn`(or `npm install`) worked for me

This worked for me as of just now

**Comment by ganeshtak19** [comment]
![Uploading ImportedPhoto_1728366053966.jpg…]()
I am facing this issue 
TypeError: Cannot read property 'config' of undefined, js engine: hermes


"react-native-gesture-handler": "2.16.2",
   "react-native-reanimated": "^3.15.4",

I have tried all above solutions but none of them worked for me.

**Comment by zubyrbutt** [comment]
> The error is very dynamic. Check your main configuration files that everything is ok. In my case, in the `.env` I had an incorrect value that was needed for the state of the app.

this worked for me. missing .env files in my project root

**Comment by KarthikShetty5** [comment]
ERROR  TypeError: Cannot read property 'ARMA_RSSI_FILTER' of null, js engine: hermes
 ERROR  Invariant Violation: "main" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes

can anyone help me with this
